### PR TITLE
feat: Support count AggregateUDF for window function

### PR DESCRIPTION
**Pull request by huaxingao**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion-comet/issues/645.

### CODE REVIEW

**Comment by huaxingao** [comment]
cc @andygrove @kazuyukitanimura @viirya

File: spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala [hunk 1]
```diff
@@ -211,7 +211,7 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
             // TODO add support for Count (this was removed when upgrading
             // to DataFusion 40 because it is no longer a built-in window function)
             // https://github.com/apache/datafusion-comet/issues/645
```
Comment (andygrove) [inline spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala:213]: This comment can be removed as part of this PR

```suggestion
```

File: spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala [hunk 2]
```diff
@@ -1466,7 +1466,7 @@ class CometExecSuite extends CometTestBase {
         SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> aqeEnabled) {
         withParquetTable((0 until 10).map(i => (i, 10 - i)), "t1") { // TODO: test nulls
           val aggregateFunctions =
-            List("MAX(_1)", "MIN(_1)") // TODO: Test all the aggregates
+            List("COUNT(_1)", "MAX(_1)", "MIN(_1)") // TODO: Test all the aggregates
```
Comment (andygrove) [inline spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala:1469]: Is `count(*)` also valid in this context? Could you add a test if it is

File: native/core/src/execution/datafusion/planner.rs [hunk 3]
```diff
@@ -1599,6 +1602,24 @@ impl PhysicalPlanner {
         }
     }
 
+    /// Find DataFusion's built-in window function by name.
+    fn find_df_window_func(name: &str) -> Option<WindowFunctionDefinition> {
```
Comment (kazuyukitanimura) [inline native/core/src/execution/datafusion/planner.rs:1606]: Is it possible to reuse the original `find_df_window_func`?
E.g.
```
if let Some(f) = find_df_window_func(name) {
             Some(f)
} else {
              match name.as_str() {
                 "count" => Some(WindowFunctionDefinition::AggregateUDF(count_udaf())),
                 _ => None,
             }
}
```

File: spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala (hunk 1 above)
Comment (huaxingao) [reply spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala]: Removed. Thanks

File: spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala (hunk 2 above)
Comment (huaxingao) [reply spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala]: Added. Thanks

File: native/core/src/execution/datafusion/planner.rs (hunk 3 above)
Comment (huaxingao) [reply native/core/src/execution/datafusion/planner.rs]: Sounds good. Changed.

**Review by andygrove — APPROVED** [review]
LGTM. Thanks @huaxingao

File: native/core/src/execution/datafusion/planner.rs [hunk 4]
```diff
@@ -1599,6 +1599,18 @@ impl PhysicalPlanner {
         }
     }
 
+    /// Find DataFusion's built-in window function by name.
+    fn find_df_window_function(name: &str) -> Option<WindowFunctionDefinition> {
+        if let Some(f) = find_df_window_func(name) {
+            Some(f)
+        } else {
+            match name {
```
Comment (kazuyukitanimura) [inline native/core/src/execution/datafusion/planner.rs:1607]: Do we need to do `name.to_lowercase()`?

Comment (huaxingao) [reply native/core/src/execution/datafusion/planner.rs:1607]: I think we are OK. `name` is from this [method](https://github.com/apache/datafusion-comet/blob/04ceba8661b4e402a41586c51c8f340389fb8116/native/core/src/execution/datafusion/planner.rs#L1588). I have lower case `count`, `min`, `max` for all of them.

**Review by kazuyukitanimura — APPROVED** [review]
LGTM

File: native/core/src/execution/datafusion/planner.rs [hunk 5]
```diff
@@ -1599,6 +1599,18 @@ impl PhysicalPlanner {
...
+                "count" => Some(WindowFunctionDefinition::AggregateUDF(count_udaf())),
+                _ => None,
```
Comment (viirya) [inline native/core/src/execution/datafusion/planner.rs:1609]: Can we look for aggregate udf from FunctionRegistry? It is similar to how we look for scalar function from `FunctionRegistry`.

Comment (viirya) [reply native/core/src/execution/datafusion/planner.rs:1609]: You can check `create_comet_physical_fun`.

Comment (huaxingao) [reply native/core/src/execution/datafusion/planner.rs]: Changed. Thanks

**Review by viirya — APPROVED** [review]

**Pull request merged**